src/floris/tools/floris_interface.py

```diff
@@ -569,82 +866,37 @@ def get_farm_power(
-
-                if unc_options["std_yaw"] > 0:
-                    yaw_bnd = int(
-                        np.ceil(
-                            norm.ppf(unc_options["pdf_cutoff"], scale=unc_options["std_yaw"]) / unc_options["pmf_res"]
-                        )
-                    )
-                    yaw_unc = np.linspace(
-                        -1 * yaw_bnd * unc_options["pmf_res"],
-                        yaw_bnd * unc_options["pmf_res"],
-                        2 * yaw_bnd + 1,
-                    )
-                    yaw_unc_pmf = norm.pdf(yaw_unc, scale=unc_options["std_yaw"])
-                    # normalize so sum = 1.0
-                    yaw_unc_pmf = yaw_unc_pmf / np.sum(yaw_unc_pmf)
-                else:
-                    yaw_unc = np.zeros(1)
-                    yaw_unc_pmf = np.ones(1)
-
-                unc_pmfs = {
-                    "wd_unc": wd_unc,
-                    "wd_unc_pmf": wd_unc_pmf,
-                    "yaw_unc": yaw_unc,
-                    "yaw_unc_pmf": yaw_unc_pmf,
-                }
+        if include_unc:
+            unc_pmfs = _generate_uncertainty_parameters(unc_options, unc_pmfs)
 
-            mean_farm_power = 0.0
-            wd_orig = self.floris.farm.wind_map.input_direction[0]
+            # TODO: The original form of this is:
+            # self.floris.farm.wind_map.input_direction[0], but it's unclear why we're
+            # capping at just the first wind direction. Should this behavior be kept?
+            # I'm unsure as to how the first wind direction is the original, so it could
+            # just be a naming thing that's throwing me off....
+            wd_orig = self.floris.flow_field.wind_directions
 
             yaw_angles = self.get_yaw_angles()
-
-            for i_wd, delta_wd in enumerate(unc_pmfs["wd_unc"]):
-                self.reinitialize_flow_field(wind_direction=wd_orig + delta_wd)
-
-                for i_yaw, delta_yaw in enumerate(unc_pmfs["yaw_unc"]):
-                    mean_farm_power = mean_farm_power + unc_pmfs["wd_unc_pmf"][i_wd] * unc_pmfs["yaw_unc_pmf"][
-                        i_yaw
-                    ] * self.get_farm_power_for_yaw_angle(list(np.array(yaw_angles) + delta_yaw), no_wake=no_wake)
+            self.reinitialize_flow_field(wind_direction=wd_orig + unc_pmfs["wd_unc"])
+            power_at_yaw = [
+                self.get_farm_power_for_yaw_angle(yaw_angles + delta_yaw, no_wake=no_wake)
+                for delta_yaw in unc_pmfs["yaw_unc"]
+            ]
+            mean_farm_power = unc_pmfs["wd_unc_pmf"] * unc_pmfs["yaw_unc_pmf"] * np.array(power_at_yaw)
 
             # reinitialize with original values
             self.reinitialize_flow_field(wind_direction=wd_orig)
             self.calculate_wake(yaw_angles=yaw_angles, no_wake=no_wake)
             return mean_farm_power
-        else:
-            turb_powers = [turbine.power for turbine in self.floris.farm.turbines]
-            return np.sum(turb_powers)
+
```

Comment:
It's an implicit else, so it reduces the number of control flow levels and ensures, we always reach the end of the function. By not having so many levels, it also makes reading a bit easier (though it could be that the reduction of levels can be confusing at first).